File: app/src/main/java/com/video/downloading/app/downloader/online/app/actvities/VideoPlayActivity.java
package com.video.downloading.app.downloader.online.app.actvities;

import android.app.ProgressDialog;
import android.content.Intent;
import android.content.pm.ActivityInfo;
import android.media.MediaPlayer;
import android.net.Uri;
import android.os.Bundle;
import android.util.DisplayMetrics;
import android.util.Log;
import android.view.KeyEvent;
import android.view.MotionEvent;
import android.view.SurfaceHolder;
import android.view.SurfaceView;
import android.view.View;
import android.widget.FrameLayout;

import androidx.appcompat.app.AppCompatActivity;


import com.video.downloading.app.downloader.online.app.R;
import com.video.downloading.app.downloader.online.app.utils.VideoControllerView;

import java.util.Objects;

import static com.video.downloading.app.downloader.online.app.utils.Constants.TAGI;

public class VideoPlayActivity extends AppCompatActivity implements SurfaceHolder.Callback, MediaPlayer.OnPreparedListener, VideoControllerView.MediaPlayerControl {
    static MediaPlayer player;
    private static boolean mFullScreen = true;
    SurfaceView videoSurface;
    VideoControllerView controller;
    String filepath;
    String fromStreaming;
    private ProgressDialog progressDialog;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_video_play);
        videoSurface = findViewById(R.id.videoSurface);
        progressDialog = new ProgressDialog(this);
        progressDialog.setMessage("Loading..");
        progressDialog.setCancelable(false);
        SurfaceHolder videoHolder = videoSurface.getHolder();
        videoHolder.addCallback(this);
        Intent i = getIntent();
        Bundle extras = i.getExtras();
        filepath = Objects.requireNonNull(extras).getString("videofilename");
        fromStreaming = extras.getString("fromStreaming");
        player = new MediaPlayer();
        controller = new VideoControllerView(this);
        View decorView = getWindow().getDecorView();
        int uiOptions = View.SYSTEM_UI_FLAG_HIDE_NAVIGATION |
                View.SYSTEM_UI_FLAG_FULLSCREEN | View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY;
        decorView.setSystemUiVisibility(uiOptions);
        try {
            if (fromStreaming == null) {
                player.setDataSource(filepath);
            } else {
                Uri video = Uri.parse(filepath);
                player.setDataSource(getApplicationContext(),video);
            }
            progressDialog.show();
            player.setOnPreparedListener(this);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    @Override
    public void onPrepared(MediaPlayer mediaPlayer) {
        controller.setMediaPlayer(this);
        int videoWidth = player.getVideoWidth();
        int videoHeight = player.getVideoHeight();
        float videoProportion = (float) videoWidth / (float) videoHeight;
        int screenWidth = getWindowManager().getDefaultDisplay().getWidth();
        int screenHeight = getWindowManager().getDefaultDisplay().getHeight();
        float screenProportion = (float) screenWidth / (float) screenHeight;
        android.view.ViewGroup.LayoutParams lp = videoSurface.getLayoutParams();

        if (videoProportion > screenProportion) {
            lp.width = screenWidth;
            lp.height = (int) ((float) screenWidth / videoProportion);
        } else {
            lp.width = (int) (videoProportion * (float) screenHeight);
            lp.height = screenHeight;
        }
        videoSurface.setLayoutParams(lp);
        controller.setAnchorView((FrameLayout) findViewById(R.id.videoSurfaceContainer));
        progressDialog.dismiss();
        player.start();
    }

    @Override
    public void surfaceCreated(SurfaceHolder surfaceHolder) {
        Log.d(TAGI,"surfaceCreated: ");
        player.setDisplay(surfaceHolder);
        player.prepareAsync();
    }

    @Override
    public void surfaceChanged(SurfaceHolder surfaceHolder,int i,int i1,int i2) {
        Log.d(TAGI,"surfaceChanged: ");
    }

    @Override
    public void surfaceDestroyed(SurfaceHolder surfaceHolder) {
        Log.d(TAGI,"surfaceDestroyed: ");
    }

    @Override
    public void start() {
        Log.d(TAGI,"start: ");
        player.start();
    }

    @Override
    public void pause() {
        Log.d(TAGI,"pause: ");
        player.pause();
    }

    @Override
    public int getDuration() {
        if (player != null) {
            Log.d(TAGI,"getDuration: ");
            return player.getDuration();
        } else {
            Log.d(TAGI,"getDuration: 0");
            return 0;
        }
    }

    @Override
    public int getCurrentPosition() {
        if (player != null) {
            Log.d(TAGI,"getCurrentPosition: ");
            return player.getCurrentPosition();
        } else {
            Log.d(TAGI,"getCurrentPosition: 0");
            return 0;
        }
    }

    @Override
    public void seekTo(int pos) {
        Log.d(TAGI,"seekTo: " + pos);
        player.seekTo(pos);
    }

    @Override
    public boolean isPlaying() {
        if (player != null) {
           // progressDialog.dismiss();
            Log.d(TAGI,"isPlaying: 1 ");
            return player.isPlaying();
        } else {
        //    progressDialog.dismiss();
            Log.d(TAGI,"isPlaying: 2");
            return false;
        }
    }

    @Override
    public int getBufferPercentage() {
//        progressDialog.show();
        Log.d(TAGI,"getBufferPercentage: ");
        return 0;
    }

    @Override
    public boolean canPause() {
        return true;
    }

    @Override
    public boolean canSeekBackward() {
        Log.d(TAGI,"canSeekBackward: ");
        return true;
    }

    @Override
    public boolean canSeekForward() {
        Log.d(TAGI,"canSeekForward: ");
        return true;
    }

    @Override
    public void stop() {
        Log.d(TAGI,"stop: ");
        player.stop();
        player.release();
    }

    @Override
    public boolean isFullScreen() {
        if (mFullScreen) {
            Log.d(TAGI,"--set icon full screen--");
            return false;
        } else {
            Log.d(TAGI,"--set icon small full screen--");
            return true;
        }
    }

    public void setFullScreen(boolean fullScreen) {
        fullScreen = false;

        if (mFullScreen) {
            Log.d(TAGI,"-----------Set full screen SCREEN_ORIENTATION_LANDSCAPE------------");
            setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE);
            DisplayMetrics displaymetrics = new DisplayMetrics();
            getWindowManager().getDefaultDisplay().getMetrics(displaymetrics);
            int height = displaymetrics.heightPixels;
            int width = displaymetrics.widthPixels;
            FrameLayout.LayoutParams params = (FrameLayout.LayoutParams) videoSurface.getLayoutParams();
            params.width = width;
            params.height = height;
            params.setMargins(0,0,0,0);
            //set icon is full screen
            mFullScreen = fullScreen;
        } else {
            Log.d(TAGI,"-----------Set small screen SCREEN_ORIENTATION_PORTRAIT------------");
            setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);
            DisplayMetrics displaymetrics = new DisplayMetrics();
            getWindowManager().getDefaultDisplay().getMetrics(displaymetrics);
            final FrameLayout mFrame = findViewById(R.id.videoSurfaceContainer);
            // int height = displaymetrics.heightPixels;
            int height = mFrame.getHeight();//get height Frame Container video
            int width = displaymetrics.widthPixels;
            FrameLayout.LayoutParams params = (FrameLayout.LayoutParams) videoSurface.getLayoutParams();
            params.width = width;
            params.height = height;
            params.setMargins(0,0,0,0);
            //set icon is small screen
            mFullScreen = true;
        }
    }

    @Override
    public void toggleFullScreen() {
        Log.d(TAGI,"-----------------click toggleFullScreen-----------");
        setFullScreen(isFullScreen());
    }

    @Override
    public boolean onTouchEvent(MotionEvent event) {
        controller.show();
        Log.d(TAGI,"onTouchEvent: ");
        return false;
    }

    @Override
    public boolean onKeyDown(int keyCode,KeyEvent event) {
        if ((keyCode == KeyEvent.KEYCODE_BACK)) { //Back key pressed
            //Things to Do
            Log.d(TAGI,"onKeyDown: ");
            if (player != null) {
                player.stop();
                player = null;
            }
            finish();
            return true;
        }
        return super.onKeyDown(keyCode,event);
    }
}

File: app/src/main/java/com/video/downloading/app/downloader/online/app/actvities/VideoPlayActivity.kt
package com.video.downloading.app.downloader.online.app.actvities

import android.content.pm.ActivityInfo
import android.media.MediaPlayer
import android.media.MediaPlayer.OnPreparedListener
import android.net.Uri
import android.os.Bundle
import android.util.DisplayMetrics
import android.util.Log
import android.view.*
import android.widget.FrameLayout
import com.video.downloading.app.downloader.online.app.R
import com.video.downloading.app.downloader.online.app.utils.Constants.TAGI
import com.video.downloading.app.downloader.online.app.utils.VideoControllerView

@Suppress("VARIABLE_WITH_REDUNDANT_INITIALIZER", "DEPRECATION")
class VideoPlayActivity : BaseActivity(), SurfaceHolder.Callback,
    OnPreparedListener, VideoControllerView.MediaPlayerControl {
    private var videoSurface: SurfaceView? = null
    private var controller: VideoControllerView? = null
    private var filepath: String? = null
    private var fromStreaming: String? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_video_play)
        videoSurface = findViewById(R.id.videoSurface)

        val videoHolder = videoSurface!!.holder
        videoHolder.addCallback(this)
        val i = intent
        val extras = i.extras
        filepath = extras!!.getString("videofilename")
        fromStreaming = extras.getString("fromStreaming")
        player = MediaPlayer()
        controller = VideoControllerView(this)
        val decorView = window.decorView
        val uiOptions = View.SYSTEM_UI_FLAG_HIDE_NAVIGATION or
                View.SYSTEM_UI_FLAG_FULLSCREEN or View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY
        decorView.systemUiVisibility = uiOptions
        try {
            if (fromStreaming == null) {
                player!!.setDataSource(filepath)
            } else {
                val video = Uri.parse(filepath)
                player!!.setDataSource(applicationContext, video)
            }
            showDialog("Loading..")
            player!!.setOnPreparedListener(this)
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    override fun onPrepared(mediaPlayer: MediaPlayer) {
        controller!!.setMediaPlayer(this)
        val videoWidth = player!!.videoWidth
        val videoHeight = player!!.videoHeight
        val videoProportion = videoWidth.toFloat() / videoHeight.toFloat()
        val screenWidth = windowManager.defaultDisplay.width
        val screenHeight = windowManager.defaultDisplay.height
        val screenProportion =
            screenWidth.toFloat() / screenHeight.toFloat()
        val lp = videoSurface!!.layoutParams
        if (videoProportion > screenProportion) {
            lp.width = screenWidth
            lp.height = (screenWidth.toFloat() / videoProportion).toInt()
        } else {
            lp.width = (videoProportion * screenHeight.toFloat()).toInt()
            lp.height = screenHeight
        }
        videoSurface!!.layoutParams = lp
        controller!!.setAnchorView(findViewById<View>(R.id.videoSurfaceContainer) as FrameLayout)
        hideDialog()
        player!!.start()
    }

    override fun surfaceCreated(surfaceHolder: SurfaceHolder) {
        Log.d(TAGI, "surfaceCreated: ")
        player!!.setDisplay(surfaceHolder)
        player!!.prepareAsync()
    }

    override fun surfaceChanged(
        surfaceHolder: SurfaceHolder,
        i: Int,
        i1: Int,
        i2: Int
    ) {
        Log.d(TAGI, "surfaceChanged: ")
    }

    override fun surfaceDestroyed(surfaceHolder: SurfaceHolder) {
        Log.d(TAGI, "surfaceDestroyed: ")
    }

    override fun start() {
        Log.d(TAGI, "start: ")
        player!!.start()
    }

    override fun pause() {
        Log.d(TAGI, "pause: ")
        player!!.pause()
    }

    override fun getDuration(): Int {
        return if (player != null) {
            Log.d(TAGI, "getDuration: ")
            player!!.duration
        } else {
            Log.d(TAGI, "getDuration: 0")
            0
        }
    }

    override fun getCurrentPosition(): Int {
        return if (player != null) {
            Log.d(TAGI, "getCurrentPosition: ")
            player!!.currentPosition
        } else {
            Log.d(TAGI, "getCurrentPosition: 0")
            0
        }
    }

    override fun seekTo(pos: Int) {
        Log.d(TAGI, "seekTo: $pos")
        player!!.seekTo(pos)
    }

    override fun isPlaying(): Boolean {
        return if (player != null) {
            // progressDialog.dismiss();
            Log.d(TAGI, "isPlaying: 1 ")
            player!!.isPlaying
        } else {
            //    progressDialog.dismiss();
            Log.d(TAGI, "isPlaying: 2")
            false
        }
    }

    override fun getBufferPercentage(): Int {
//        progressDialog.show();
        Log.d(TAGI, "getBufferPercentage: ")
        return 0
    }

    override fun canPause(): Boolean {
        return true
    }

    override fun canSeekBackward(): Boolean {
        Log.d(TAGI, "canSeekBackward: ")
        return true
    }

    override fun canSeekForward(): Boolean {
        Log.d(TAGI, "canSeekForward: ")
        return true
    }

    override fun stop() {
        Log.d(TAGI, "stop: ")
        player!!.stop()
        player!!.release()
    }

    override fun isFullScreen(): Boolean {
        return if (mFullScreen) {
            Log.d(TAGI, "--set icon full screen--")
            false
        } else {
            Log.d(TAGI, "--set icon small full screen--")
            true
        }
    }

    fun setFullScreen(fullScreen: Boolean) {
        var fullScreen1 = fullScreen
        fullScreen1 = false
        if (mFullScreen) {
            Log.d(
                TAGI,
                "-----------Set full screen SCREEN_ORIENTATION_LANDSCAPE------------"
            )
            requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE
            val displaymetrics = DisplayMetrics()
            windowManager.defaultDisplay.getMetrics(displaymetrics)
            val height = displaymetrics.heightPixels
            val width = displaymetrics.widthPixels
            val params =
                videoSurface!!.layoutParams as FrameLayout.LayoutParams
            params.width = width
            params.height = height
            params.setMargins(0, 0, 0, 0)
            //set icon is full screen
            mFullScreen = fullScreen1
        } else {
            Log.d(
                TAGI,
                "-----------Set small screen SCREEN_ORIENTATION_PORTRAIT------------"
            )
            requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
            val displaymetrics = DisplayMetrics()
            windowManager.defaultDisplay.getMetrics(displaymetrics)
            val mFrame = findViewById<FrameLayout>(R.id.videoSurfaceContainer)
            // int height = displaymetrics.heightPixels;
            val height = mFrame.height //get height Frame Container video
            val width = displaymetrics.widthPixels
            val params =
                videoSurface!!.layoutParams as FrameLayout.LayoutParams
            params.width = width
            params.height = height
            params.setMargins(0, 0, 0, 0)
            //set icon is small screen
            mFullScreen = true
        }
    }

    override fun toggleFullScreen() {
        Log.d(TAGI, "-----------------click toggleFullScreen-----------")
        isFullScreen = isFullScreen
    }

    override fun onTouchEvent(event: MotionEvent): Boolean {
        controller!!.show()
        Log.d(TAGI, "onTouchEvent: ")
        return false
    }

    override fun onKeyDown(keyCode: Int, event: KeyEvent): Boolean {
        if (keyCode == KeyEvent.KEYCODE_BACK) { //Back key pressed
            //Things to Do
            Log.d(TAGI, "onKeyDown: ")
            if (player != null) {
                player!!.stop()
                player = null
            }
            finish()
            return true
        }
        return super.onKeyDown(keyCode, event)
    }

    companion object {
        var player: MediaPlayer? = null
        private var mFullScreen = true
    }
}